### ISSUE TITLE
Build failed with openh264 codec

### ISSUE THREAD

**Issue by jakubprusa** [BelledonneCommunications/linphone-iphone#19]
Hi, 

I am trying to build linphone with openh264 codec and failed with 

Run command:
```
./prepare.py -c && ./prepare.py --enable-non-free-codecs -DENABLE_OPENH264=ON all && make
```

Output:

```
-- Installing: /Users/ios/linphone-iphone/liblinphone-sdk/x86_64-apple-darwin.ios/lib/mediastreamer/plugins/libmsilbc.a
[100%] Completed 'EP_msilbc'
[100%] Built target EP_msilbc
rm -f WORK/ios-x86_64/Stamp/EP_msopenh264/EP_msopenh264-update; \
  /Applications/Xcode.app/Contents/Developer/usr/bin/make -C WORK/ios-x86_64/cmake EP_msopenh264
Scanning dependencies of target EP_openh264
[  0%] Creating directories for 'EP_openh264'
[  1%] No download step for 'EP_openh264'
[  3%] No patch step for 'EP_openh264'
[  3%] No update step for 'EP_openh264'
[  3%] Checking for source code changes for 'EP_openh264'
[  4%] Generating /Users/ios/linphone-iphone/WORK/ios-x86_64/Build/openh264/check_changes.txt
[  4%] Performing configure step for 'EP_openh264'
[  6%] Performing build step for 'EP_openh264'
/Users/ios/linphone-iphone/submodules/externals/openh264//codec/common/x86/deblock.asm:1758: fatal: No section for index 2 offset 0 found
make[5]: *** [codec/common/x86/deblock.o] Error 1
make[4]: *** [/Users/ios/linphone-iphone/WORK/ios-x86_64/Stamp/EP_openh264/EP_openh264-build] Error 2
make[3]: *** [CMakeFiles/EP_openh264.dir/all] Error 2
make[2]: *** [CMakeFiles/EP_msopenh264.dir/rule] Error 2
make[1]: *** [EP_msopenh264] Error 2
make: *** [x86_64-build-msopenh264] Error 2
```

Full log here: [error_log.txt](https://github.com/BelledonneCommunications/linphone-iphone/files/33004/error_log.txt)

**Comment by bagage** [comment]
Which commit are you using (`git describe`). I tried here and it worked flawlessly on latest commit 935163f. If running latest version, can you do a last test: `git submodule sync && git submodule update --init --recursive` then `./prepare.py -c && ./prepare.py --enable-non-free-codecs -DENABLE_OPENH264=ON all && make`?

**Comment by jakubprusa** [comment]
I tried as you described but result is same. I sending you full output from console in attachment
[error_log_2.txt](https://github.com/BelledonneCommunications/linphone-iphone/files/34003/error_log_2.txt)
.

**Comment by bagage** [comment]
What's your output of `nasm -v`?

**Comment by jakubprusa** [comment]
Hi,

NASM version 2.11.08 compiled on Sep 22 2015

**Comment by bagage** [comment]
I've no clue for this yet and I do not reproduce the issue.. any help welcome!

**Comment by jakubprusa** [comment]
And what is your version of nasm? I have this as virtual machine for Parallels so i can send you the whole virtual :)

**Comment by bagage** [comment]
Same as yours. Installed with brew as well.

**Comment by jakubprusa** [comment]
And do you have parallels?

**Comment by bagage** [comment]
No, sorry.

**Comment by jakubprusa** [comment]
Hmm, 

I made few experiments and now it seems that i made successful build. 

I made these steps:
```
brew tap homebrew/versions 
brew search nasm
brew install nasm21106 
./prepare.py -c && ./prepare.py --enable-non-free-codecs -DENABLE_OPENH264=ON all && make
```

I will test it and I will let you know with the final result, if we can close this issue.

**Comment by bagage** [comment]
@jakubprusa any news?

**Comment by jakubprusa** [comment]
@bagage sorry, I forgot to write latest news, as apache sais: IT WORKS !

**Comment by fisky2014** [comment]
i cant able to build with 
/prepare.py -DENABLE_NON_FREE_CODECS=ON -DENABLE_OPENH264=ON arm64  all  && make. could any one help me

**Comment by bagage** [comment]
Openh264 + xcode 7.3 not working yet.

**Comment by bagage** [comment]
See https://github.com/cisco/openh264/issues/2434, which is the upstream problem here.

**Comment by bagage** [comment]
Fixed in 95886b4 thanks to upstream fix.
